lib/mongo/auth/stringprep.rb

```diff
@@ -2,9 +2,29 @@
 
 module Mongo
   module Auth
+    # This namespace contains all behavior related to string preparation (RFC 3454).
```

Comment:
Maybe add something about how this is implemented to support scram sha 256, available in versions at least 4.0 of the mongodb server.